payments-core/src/main/java/com/stripe/android/payments/core/authentication/DefaultIntentAuthenticatorRegistry.kt

```diff
@@ -0,0 +1,150 @@
+        }
+
+        when (val nextActionData = stripeIntent.nextActionData) {
+            is StripeIntent.NextActionData.SdkData.Use3DS2 -> {
+                return stripe3DS2Authenticator
+            }
+            is StripeIntent.NextActionData.SdkData.Use3DS1 -> {
+                // can only triggered when `use_stripe_sdk=true`
+                return webIntentAuthenticator
+            }
+            is StripeIntent.NextActionData.RedirectToUrl -> {
+                // can only triggered when `use_stripe_sdk=false`
+                return webIntentAuthenticator
+            }
+            is StripeIntent.NextActionData.AlipayRedirect -> {
+                return webIntentAuthenticator
+            }
+            is StripeIntent.NextActionData.DisplayOxxoDetails -> {
+                return webIntentAuthenticator.takeIf { nextActionData.hostedVoucherUrl != null }
+                    ?: noOpIntentAuthenticator
+            }
+            else -> return noOpIntentAuthenticator
+        }
+    }
+
+    companion object {
+        // Holding this single instance would in turn holds DaggerAuthenticationComponent instance,
+        // which keeps dagger injection graph live.
+        @Volatile
+        private var INSTANCE: IntentAuthenticatorRegistry? = null
+
+        /**
+         * Return the singleton instance of [IntentAuthenticatorRegistry].
+         */
+        fun getInstance(
+            stripeRepository: StripeRepository,
+            paymentRelayStarterFactory: (AuthActivityStarter.Host) -> PaymentRelayStarter,
+            paymentBrowserAuthStarterFactory: (AuthActivityStarter.Host) -> PaymentBrowserAuthStarter,
+            analyticsRequestExecutor: AnalyticsRequestExecutor,
+            analyticsRequestFactory: AnalyticsRequestFactory,
+            logger: Logger,
+            enableLogging: Boolean,
+            workContext: CoroutineContext,
+            uiContext: CoroutineContext,
+            threeDs2Service: StripeThreeDs2Service,
+            messageVersionRegistry: MessageVersionRegistry,
+            challengeProgressActivityStarter: StripePaymentController.ChallengeProgressActivityStarter,
+            stripe3ds2Config: PaymentAuthConfig.Stripe3ds2Config,
+            stripe3ds2ChallengeLauncher: ActivityResultLauncher<PaymentFlowResult.Unvalidated>?
+        ) = INSTANCE ?: synchronized(this) {
+            INSTANCE ?: createInstance(
+                stripeRepository,
+                paymentRelayStarterFactory,
+                paymentBrowserAuthStarterFactory,
+                analyticsRequestExecutor,
+                analyticsRequestFactory,
+                logger,
+                enableLogging,
+                workContext,
+                uiContext,
+                threeDs2Service,
+                messageVersionRegistry,
+                challengeProgressActivityStarter,
+                stripe3ds2Config,
+                stripe3ds2ChallengeLauncher
+            ).also { INSTANCE = it }
```

Comment:
nit: looks like `createInstance` is already assigning it?

Reply:
you're right, updated the `createInstance` impl to not assign instance